RESUMABLE: Describe GET request against upload resource (#3467)
httpwg/http-extensions: draft-ietf-httpbis-resumable-upload.md

### Client Behavior

> If the client wants to resume the upload after an interruption, it has to know the amount of representation data processed by the upload resource so far. It can fetch the offset by sending a `HEAD` or `GET` request to the upload resource. Using `HEAD` is RECOMMENDED as no response content is needed to resume. Upon a successful response, the client can continue the upload by appending representation data ({{upload-appending}}) starting at the offset indicated by the `Upload-Offset` response header field.

Reviewer: nit:

```suggestion
If the client wants to resume the upload after an interruption, it has to know the amount of representation data processed by the upload resource so far. It can fetch the offset by sending a `HEAD` or `GET` request to the upload resource. Using `HEAD` is RECOMMENDED, since response content is not required for resumption. Upon a successful response, the client can continue the upload by appending representation data ({{upload-appending}}) starting at the offset indicated by the `Upload-Offset` response header field.
```

The resource SHOULD NOT generate a response with the `301 (Moved Permanently)` and `302 (Found)` status codes because clients might follow the redirect without preserving the `HEAD` method.

> This document does not place any requirements on the response content for `GET` requests. In particular, the server is not required to respond with the transferred representation data.

Reviewer 2: I do like this being clarified personally, because I would have expected the opposite if I were reading this the first time. But can we say "expected" instead of "required" for just the second sentence so it sounds less like servers may be inclined to respond with the transferred representation data by convention?

```suggestion
This document does not place any requirements on the response content for `GET` requests. In particular, the server is not expected to respond with the transferred representation data.
```

> This document does not place any requirements on the response content for `GET` requests. In particular, the server is not required to respond with the transferred representation data.

Reviewer: In addition to Grant's comment (which I tend to agree with), we can perhaps reword things slightly. I won't try to combine my suggestion with Grant's (because of GitHub UI), don't read that as me disagreeing. The following is shared to illustrate my point. Something like:

```suggestion
A client does not require response content for an offset retrieval request in order to successfully resume an upload.  Therefore, serving response content for a `GET` request is unexpected. Its meaning is not defined by this protocol.
```

Reviewer 2: I like this better! Though you may want to fix the double spaces before "Therefore".

Author: ```suggestion
A client does not require response content for an offset retrieval request in order to successfully resume an upload. Therefore, serving response content for a `GET` request is unexpected. Its meaning is not defined by this protocol.
```
